src/main/java/com/amihaiemil/eoyaml/ReadPlainScalar.java

```diff
@@ -35,6 +38,17 @@
  */
 final class ReadPlainScalar extends BaseScalar {
 
+    /**
+     * Pattern to match quoted literals, mappings or sequences.
+     */
+    private static final Pattern QUOTED_LITERAL_MAP_SEQ = Pattern.compile(
+            "^(([ ]*([\\-][ ]*)?)"
+            + "((\"(?:[^\"\\\\]|\\\\.)*\")|"
+            + "('(?:[^'\\\\]|\\\\.)*'))|"
+            + "([ ]*(.*)(:|:[ ]+(.*)))|"
+            + "(-[ ]+(.*))|"
+            + ")$");
```

Comment:
@newmana Same here, detailed JavaDoc is needed, because nobody will have any idea what this regex does D:

Reply:
Done for both.